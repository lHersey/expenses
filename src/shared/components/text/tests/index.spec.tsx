import { render } from '@testing-library/react-native';
import React, { ComponentProps, FC } from 'react';
import Providers from 'shared/providers';

import BaseText from '../index';

const Text: FC<ComponentProps<typeof BaseText>> = props => {
  return (
    <Providers>
      <BaseText {...props} />
    </Providers>
  );
};

describe('<Text />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(<Text>Hello Text</Text>).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const { queryByText } = render(<Text>Hello Text</Text>);

      expect(queryByText('Hello Text')).toBeTruthy();
    });
  });
});
