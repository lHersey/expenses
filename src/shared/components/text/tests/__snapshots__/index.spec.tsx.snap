// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> snapshots should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3e3e3e",
        "fontSize": 24,
      },
    ]
  }
>
  Hello Text
</Text>
`;
